### ISSUE TITLE
Configuration option to ignore failures

### ISSUE THREAD

**Issue by damianw** [runningcode/fladle#48]
Standard test tasks support `ignoreFailures`, which Flank doesn't appear to. Ignoring failures is useful for CI, where it's more desirable for Gradle invocations to complete successfully, and test results to be analyzed afterward.

**Comment by runningcode** [comment]
The flank task doesn't extend from the standard test task so it doesn't have the `ignoreFailures` property. Would running gradle with the `--continue` flag solve your use case? More information here: https://docs.gradle.org/current/userguide/command_line_interface.html#sec:continue_build_on_failure

**Comment by damianw** [comment]
Thanks for the reply.

I think `--continue` gets most of the way there, but it has a couple problems from what I understand. For one, it applies to all tasks, not just flank tasks. It is somewhat desirable to have other kinds of tasks fail the build eagerly, but flank+test tasks to not. It also still results in a failure (non-zero exit code) from the gradle invocation... there's no clean way to separate out tests failing from an actual failed build.

That being said, I'm not really a Gradle expert, so maybe there is a way to ignore failures for specific tasks in a way that doesn't fail the build?

**Comment by runningcode** [comment]
Can you try something like this?
```
afterEvaluate {
    tasks.named("execFlank").configure {
        ignoreExitValue true
    }
}
```

**Comment by bootstraponline** [comment]
Thanks for providing that work around @runningcode I'll test it out and let you know.

**Comment by bootstraponline** [comment]
I tried this in `build.gradle.kts`

```kotlin
afterEvaluate {
    tasks.named("execFlank").configure {
        ignoreExitValue = true
    }
}
```

I see:

```
Line 321:         ignoreExitValue = true
                    ^ Unresolved reference: ignoreExitValue
```

Maybe the execFlank task doesn't support `ignoreExitValue`?

It may be easier for me to add `ignoreFailures` directly to Flank. The use case as I understand it, is that Jenkins will look at the JUnit XML for the results. The exit code for the build should be green if the build was created successfully, regardless of the test outcome.

It seems weird but apparently `ignoreFailures` is a thing in Gradle.

**Comment by runningcode** [comment]
Kotlin needs a bit of help to figure out the type of the `execFlank` task.
Please try the following kts compatible™ snippet and let me know how it goes. :)
```
afterEvaluate {
  tasks.named("execFlank", JavaExec::class.java).configure {
    isIgnoreExitValue = true
  }
}
```

**Comment by bootstraponline** [comment]
Thanks! That worked.

**Comment by runningcode** [comment]
Woo, I finally get to close this issue!

**Comment by bootstraponline** [comment]
> Task :instrumentation-tests-merged:execFlank FAILED
> BUILD FAILED 

It looks like this feature has to be in Flank upstream. Gradle is still marking the task as failed which fails the build, even though the exit value is ignored.

**Comment by bootstraponline** [comment]
The kotlin syntax did compile successfully though. 😄

**Comment by runningcode** [comment]
Awww, just for my sanity, can you try this syntax?
```
afterEvaluate {
  tasks.named("execFlank", JavaExec::class.java).configure {
    setIgnoreExitValue(true)
  }
}
```

**Comment by bootstraponline** [comment]
Sure, I submitted that syntax to CI. I'll let you know what happens.

**Comment by runningcode** [comment]
[This works for sure.](https://github.com/runningcode/fladle/issues/48#issuecomment-533007002) I tested it locally.

**Comment by bootstraponline** [comment]
You're right, I confirmed it works. There were multiple `execFlank` tasks, it looks like one per module. Once I defined the configuration option on the correct module, it worked as expected.

Thanks for your help!

**Comment by runningcode** [comment]
In case anyone stumbles into this Github issue in the future. I would not recommend ignoring flanks exit value. If a matrix fails for any reason other than the test itself failing (bad matrix, device failure, infrastructure overload) than the failure will not appear in the junit report. This is a dangerous false positive situation in the test reporting system in flank.
See this issue for more details: https://github.com/TestArmada/flank/issues/457

**Comment by bootstraponline** [comment]
I agree with @runningcode  The exit code is the only officially supported way from Firebase Test Lab to tell if a run has passed or not. For any number of infrastructure errors, a JUnit XML file may not be generated at all. Sometimes the generated XML file itself is empty.

tl;dr check the exit code 😄

**Comment by runningcode** [comment]
Gradle 6.1 is released with the ability to check the exit code for JavaExec tasks!
https://docs.gradle.org/6.1/release-notes.html

https://github.com/gradle/gradle/pull/10763

**Comment by gildor** [comment]
@runningcode Should ignoreExitValue be a part of flank extension?

**Comment by runningcode** [comment]
I'm not sure I want to expose the internals of Gradle to flank's api. But if there's a strong use case for it, please open a PR or github issue and let's disuss there. I'm not sure I completly understand the use case here.

**Comment by gildor** [comment]
@runningcode Our use case is run fladle task on CI and do not fail of some of the tests are failed, but instead, collect Junit report and fail or not depending on results.
It's the same as Gradle test task is working with `ignoreFailures`, or Android lint with `android.lintOptions.abortOnError`.

So idea is to allow configure should a failed test fail the build. It could be a part of a bigger change in error handling. For example the task should fail if there is an infrastructure error and not fail if a test is not passed. Probably it can be done in Flank itself, to return different exit codes (and Fladle could handle those exit codes according to config)

**Comment by bootstraponline** [comment]
I reopened https://github.com/Flank/flank/issues/602

Current idea is if ignoreExitValue is set then we'll always return a 0 exit code.
